lib/crontab/date_helper.ex

```diff
@@ -202,33 +158,7 @@ defmodule Crontab.DateHelper do
 
   """
   @spec dec_year(date) :: date when date: date
-  def dec_year(date = %NaiveDateTime{}) do
-    leap_year? =
-      date
-      |> NaiveDateTime.to_date()
-      |> Date.leap_year?()
-
-    if leap_year? do
-      NaiveDateTime.add(date, -366, :day)
-    else
-      NaiveDateTime.add(date, -365, :day)
-    end
-  end
-
-  def dec_year(date = %DateTime{}) do
-    leap_year? =
-      date
-      |> DateTime.to_date()
-      |> Date.leap_year?()
-
-    if leap_year? do
-      # FIXME: How to correct date with tz?
-      DateTime.add(date, -366, :day)
-    else
-      # FIXME: How to correct date with tz?
-      DateTime.add(date, -365, :day)
-    end
-  end
+  def dec_year(date), do: add(date, -days_in_year(date), :day)
```

Comment:
We might have a bug here. (Also in the implementation before.)

Given date `2024-02-01` (before leap day), I think we subtract one day too much.

Same with increment if we're after the leap day.